### ISSUE TITLE
How to run with standalone 

### ISSUE THREAD

**Issue by srikrishnacancun** [aws-samples/data-pipeline-samples#57]
If I have my standalone spark cluster with hdfs/yarn configured , What changes are required to run this code?

**Comment by mbeitchman** [comment]
HI, 

Can you tell me which sample you are referring to? 

Is your standalone cluster an EMR cluster?

**Comment by srikrishnacancun** [comment]
Hi

I am referring to HadoopTerasort . Yes I want to run against my own
Standalone Spark cluster or Hadoop cluster. What  needs to be modified if
any to make it work . We want the output to S3 like in the example. How big
of a file size we can process.?

https://github.com/awslabs/data-pipeline-samples/tree/master/samples/HadoopTerasort

Thanks

Srikrishna

**Comment by mbeitchman** [comment]
Hi Srikrishna,

You will need to run the taskrunner on your cluster. Please see this link for more details.

http://docs.aws.amazon.com/datapipeline/latest/DeveloperGuide/dp-using-task-runner.html

I think you can process as much as you want. Of course, runtime will depend on your cluster size.

Marc

**Comment by srikrishnacancun** [comment]
Hi

Thanks for your quick response .  Can you explain the high level steps so
that I can understand ?

I don't want to use EMR Cluster .  I have a stand alone spark cluster 1.6.1
. I want to write the i/p and output to s3

Have  nice day.

Thanks

Srikrishna

**Comment by srikrishnacancun** [comment]
Hi

That means what you are saying you can replace EMR cluster with Physical
Server based Spark/Hadoop cluster ?  Is that right ?

I am very eager to receive your response.

Thanks

Srikrishna

**Comment by mbeitchman** [comment]
yes, that is correct. The task runner is an agent that runs on AWS or on premises resources to execute the activities in the pipeline. The above documentation will explain this in more detail. Please follow up if you have questions once you get started.

**Comment by srikrishnacancun** [comment]
Hi

Thanks . How do you modify your script/code to include my standalone spark
cluster installation  to create my custom pipeline .?  Can you show the
code snippet ?

srikrishna

**Comment by mbeitchman** [comment]
To connect a Task Runner that you've installed to the pipeline activities it should process, add a workerGroup field to the object, and configure Task Runner to poll for that worker group value. You do this by passing the worker group string as a parameter (for example, --workerGroup=wg-12345) when you run the Task Runner JAR file.

http://docs.aws.amazon.com/datapipeline/latest/DeveloperGuide/dp-how-task-runner-user-managed.html
